metafacture-io/src/main/java/org/metafacture/io/SruOpener.java

```diff
@@ -0,0 +1,234 @@
+        this.userAgent = userAgent;
+    }
+
+    /**
+     * Sets the query of the search.
+     * <strong>Setting a query is mandatory.</strong>
+     *
+     * @param query the query
+     */
+
+    public void setQuery(final String query) {
+        this.query = query;
+    }
+
+    /**
+     * Sets total number of records to be retrieved. <strong>Default value: indefinite (as in "all")
+     * </strong>.
+     *
+     * @param totalRecords total number of records to be retrieved
+     */
+    public void setTotal(final String totalRecords) {
+        this.totalRecords = Integer.parseInt(totalRecords);
+    }
+
+    /**
+     * Sets the maximum of records returned in one lookup. <strong>Default value: {@value MAXIMUM_RECORDS}</strong>.
+     * The lookup is repeated as long as {@link #maximumRecords} is lesser than {@link #totalRecords}.
+     *
+     * @param maximumRecords maximum of records returned in one lookup
+     */
+    public void setMaximumRecords(final String maximumRecords) {
+        this.maximumRecords = Integer.parseInt(maximumRecords);
+    }
+
+    /**
+     * Sets where to start when retrieving records. <strong>Default value: {@value START_RECORD}</strong>.
+     *
+     * @param startRecord where to start when retrieving records
+     */
+    public void setStartRecord(final String startRecord) {
+        this.startRecord = Integer.parseInt(startRecord);
+    }
+
+    /**
+     * Sets the format of the retrieved record data. <strong>Default value: {@value RECORD_SCHEMA}</strong>.
+     *
+     * @param recordSchema the format of the data of the records
+     */
+    public void setRecordSchema(final String recordSchema) {
+        this.recordSchema = recordSchema;
+    }
+
+    /**
+     * Sets the kind of operation of the lookup. <strong>Default value: {@value OPERATION}</strong>.
+     *
+     * @param operation the kind of operation of the lookup
+     */
+    public void setOperation(final String operation) {
+        this.operation = operation;
+    }
+
+    /**
+     * Sets the version of the lookup. <strong>Default value: {@value VERSION}</strong>.
+     *
+     * @param version the version of the lookup
+     */
+    public void setVersion(final String version) {
+        this.version = version;
+    }
+
+    @Override
+    public void process(final String baseUrl) {
+
+        StringBuilder srUrl = new StringBuilder(baseUrl);
+        if (query != null) {
+            srUrl.append("?query=").append(query).append("&operation=").append(operation).append("&recordSchema=")
+                .append(recordSchema).append("&version=").append(version);
+        }
+        else {
+            throw new IllegalArgumentException("Missing mandatory parameter 'query'");
+        }
+
+        while (!stopRetrieving && recordsRetrieved < totalRecords && (startRecord < numberOfRecords)) {
+            InputStream inputStream = getXmlDocsViaSru(srUrl);
+            getReceiver().process(new InputStreamReader(inputStream));
```

Comment:
What are the repercussions of passing each page as an individual document to downstream consumers? Would it make more sense to pass either the result as a whole, or each record individually?